Python/Product/Analysis/Values/IterableInfo.cs

```diff
@@ -178,7 +178,7 @@ public IterableValue(VariableDef[] indexTypes, BuiltinClassInfo seqType, Node no
         internal override bool UnionEquals(AnalysisValue ns, int strength) {
             if (strength < MergeStrength.IgnoreIterableNode) {
                 var si = ns as IterableValue;
-                if (si != null && !_node.Equals(_node)) {
```

Comment:
There's a chance that this obvious fix may unexpected side effects. Please run all tests in AnalysisTests project. We can compare failures (if any) with a branch without this fix.